include/armadillo_bits/arma_forward.hpp

```diff
@@ -280,6 +280,7 @@ struct hdf5_name
   {
   const std::string filename;
   const std::string dsname;
+  const bool append = false;
```

Comment:
This doesn't work with the older C++98 standard.  Armadillo currently supports C++98, C++11 and C++14.
The code needs to be changed to:
     
     const bool append;

and the `append` variable needs to be initialised explicitly in all constructors.